internal/node/reverse_proxy/subscriber.go

```diff
@@ -8,6 +8,25 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+func NewProcessProxyRuleStateUpdateSubscriber(ruleSet *RuleSet) (*hdb.IdempotentStateUpdateSubscriber, error) {
```

Comment:
I hate this name 😭 is there a better name or can we at least add a comment?

Reply:
I got rid of the StateUpdate part, I think we can ignore that for other ones like this going forward